vmm/src/memory_manager.rs

```diff
@@ -507,11 +531,38 @@ impl MemoryManager {
                 )
                 .map_err(Error::GuestMemory)?
             }
-            None => GuestRegionMmap::new(
-                MmapRegion::new(size).map_err(Error::GuestMemoryRegion)?,
-                start_addr,
-            )
-            .map_err(Error::GuestMemory)?,
+            None => {
+                let fd = Self::memfd_create(
+                    &ffi::CString::new("ch_ram").unwrap(),
+                    if hugepages {
+                        libc::MFD_HUGETLB | libc::MAP_HUGE_2MB as u32
```

Comment:
Should be `MAP_HUGETLB` instead of `MFD_HUGETLB`

Reply:
Nope, check the manpage for `memfd_create`

Reply:
Oh yes that's part memfd_create(), I was thinking about the mmap(), but obviously this is not needed in the mmap since the backing file is already 2M pages.